internal/writer/writer.go

```diff
@@ -124,7 +131,11 @@ func getValue(param *serializerParams, structValue reflect.Value) string {
 	case decimal.Decimal:
 		param.FillType = FillNumber
 		decimal := structValue.Interface().(decimal.Decimal)
-		return strings.ReplaceAll(decimal.StringFixed(2), ".", "")
+		if param.Precision == 0 {
```

Comment:
Esse é um caso meio ruim, pois em tese dá para colocar `precision:0` para recever apenas um inteiro. Você fez assim para evitar colocar nos outros campos o valor padrão?

Acho que seria melhor colocar em todos os campos que esse caso especial. (Por mim, tudo bem ter um caso especial se o valor a comparar não for um inteiro.)

Reply:
Aqui é uma dificuldade do Go pra mim também, pois se não preencher nada vem zero em vez de `nil` e ai é um pouco ruim fazer a comparação

Reply:
Talvez algo assim em extract_tags?

```go
func extractTags(structTagged reflect.Type) (serializerOpt serializerOpt, err error) {
	serializerOpt.Params = make([]serializerParams, structTagged.NumField())
	for i := 0; i < structTagged.NumField(); i++ {
		f := structTagged.Field(i)
		translatorTags := strings.Split(f.Tag.Get("translator"), ";")
		
		defaultPrecision bool = true // <---------------------------------------------
		for _, rule := range translatorTags {
			parts := strings.Split(rule, ";")
			for _, part := range parts {
				keyValuePair := strings.Split(part, ":")
				key := keyValuePair[0]
				value := keyValuePair[1]
				switch key {
				case "part":
					convertToInt, err := utils.ConvertStringToIntSlice(strings.Split(value, "..")...)
					if err != nil {
						return serializerOpt, err
					}
					serializerOpt.Params[i].Deliminator = convertToInt
				case "timeParse":
					serializerOpt.Params[i].TimeParse = value
				case "align":
					serializerOpt.Params[i].Align = value
				case "precision":
					precision, err := strconv.Atoi(keyValuePair[1])
					if err != nil {
						return serializerOpt, err
					}
					serializerOpt.Params[i].Precision = precision
					defaultPrecision = false // <---------------------------------------------
				}
			}

			if defaultPrecision {  // <---------------------------------------------
				serializerOpt.Params[i].Precision = 2
			}
		}
	}
	return
} 
```

Reply:
Boa, ajustei